enhancements/machine-config/custom-ignition-machineconfig.md

```diff
@@ -0,0 +1,199 @@
+
+This presents a problem for the plans to have the [MCO manage the pointer config](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/user-data-secret-managed.md)
+because it uses a static template that does not consider any user customizations, and for that reason the work was [reverted](https://github.com/openshift/machine-config-operator/pull/2126).
+
+Additionally, in some situations it is necessary to perform network configuration
+before it is possible to download the rendered config from the MCS in the ramdisk
+where ignition evaluates the pointer config.
+
+This leads to a chicken-egg problem, where a user could configure the network
+via ignition, but ignition cannot apply any config until it has downloaded
+any append/merge directives.
+
+We could solve that problem on some platforms (baremetal in particular) by
+just providing the fully rendered config to each host, bypassing the pointer
+config where config-drive size limits allow.  However this results in the [same
+issue with losing any user customizations](https://bugzilla.redhat.com/show_bug.cgi?id=1833483) provided directly via the pointer config.
+
+Discussions indicate we may want to deprecate/remove this ignition-config method of
+customization and mandate MachineConfig manifests instead.  We could in the
+meantime do that internally e.g have the installer detect any customization
+to the pointer config, and inject a MachineConfig manifest containing that data
+instead of writing it via the current user-data Secret.
+
+### Goals
+
+ * Unblock the MCO managed pointer ignition work that was previously reverted
+
+ * Provide a solution to customer requirements for baremetal IPI around bond/VLAN
+   and other configurations for controlplane networking
+
+ * Provide a means by which we might warn the user if we decide to deprecate customization via the pointer ignition config.
+
+### Non-Goals
+
+This work may give us some options wrt warnings around this interface but it
+does not aim to formally deprecate the ignition-configs customization, that
+may be separately discussed in a future enhancement.
+
+The discussion of interfaces here and how they relate to network configuration
+ only considers the existing interfaces, not proposals around a future
+[declarative network configuration](https://github.com/openshift/enhancements/pull/399) although we need to ensure the approach taken in each case is aligned.
+
+### User Stories
+
+#### Story 1
+
+As a user I would like my existing customizations to pointer ignition files
+to work, but be warned if there is a preferable interface I should be using.
+
+#### Story 2
+
+As a baremetal IPI user, I need to deploy in an environment where network
```

Comment:
Yes but...as discussed ideally we move baremetal IPI to using the Live ISO flow which already has techniques to address this by effectively having two Ignition runs.

xref https://github.com/coreos/ignition/issues/979

(OK this is covered below, thanks)

Reply:
Yes that's the long term plan and discussions are in-progress about how we may integrate the IPI baremetal automation with the assisted-install flow (which uses the live iso) - the idea with this enhancement is an interim solution so we can solve the problem for customers in 4.7 (as bond+vlans is a common requirement on baremetal)

Reply:
Also note that if you embed the pointer into the live iso that still doesn't solve the problem in the case where the MCS URL is on a tagged network, so you need to wire in keyfiles to `coreos-install --copy-network` - which then aren't managed via the MCO.

So it may be preferable in the eventual IPI+ISO flow to generate an iso on the fly with the rendered config embedded (or provided via a config drive alongside the ISO) - in which case the changes made here are still relevant.

https://github.com/metal3-io/metal3-docs/pull/150 is some early exploratory work in that direction FYI, but it's something for 4.8+